ppx/test/locations-check.t

```diff
@@ -0,0 +1,66 @@
+Test the preprocessed reason-react components have well-formed locations.
+Uses https://github.com/ocaml-ppx/ppxlib/blob/44583fc14c3cc39ee6269ffd69f52146283f72c0/src/location_check.mli
+
+With no annotations (ppx does nothing)
+
+  $ cat >input.ml <<EOF
+  > let make ~foo ~bar =
+  >   (div
+  >      ~children:[ React.string foo; bar |> string_of_int |> React.string ]
+  >      ())
+  > EOF
+
+  $ reason-react-ppx -check -locations-check input.ml
+  let make ~foo  ~bar  =
+    div ~children:[React.string foo; (bar |> string_of_int) |> React.string] ()
+
+With JSX annotation
+
+  $ cat >input.ml <<EOF
+  > let make ~foo ~bar =
+  >   (div
+  >      ~children:[ React.string foo; bar |> string_of_int |> React.string ]
+  >      () [@JSX])
+  > EOF
+
+  $ reason-react-ppx -check -locations-check input.ml
+  File "input.ml", line 2, characters 3-6:
+  2 |   (div
```

Comment:
is this expected? What exactly is this test testing if these are errors?

Reply:
or are these things we should put on our TODO list to fix?

Reply:
Yes, this should go in a TODO list to fix. The idea behind it is that if the ppx creates the AST in a way that this tests pass, we should never have any more problems with editor not finding locations, getting confused about types on hover, etc, as the tree would be consistent.

Reply:
Rather than having to write "manual" tests in a per-case basis like in https://github.com/reasonml/reason-react/pull/842.